server/neptune/workflows/internal/deploy/revision/queue/pull_rebaser.go

```
@@ -0,0 +1,159 @@
+			prsToRebase = append(prsToRebase, pullRequest)
+		}
+	}
+
+	// skip set pr revision call if no prs to rebase
+	if len(prsToRebase) == 0 {
+		return nil
+	}
+
+	// reset workflow options
+	ctx = workflow.WithActivityOptions(ctx, originalOpts)
+	futureByPullNum = map[github.PullRequest]workflow.Future{}
+	for _, pr := range prsToRebase {
+		futureByPullNum[pr] = workflow.ExecuteActivity(ctx, p.PrRevisionSetterActivities.SetPRRevision, activities.SetPRRevisionRequest{
+			Repository:  repo,
+			PullRequest: pr,
+		})
+	}
+
+	// TODO: Add alarms on SetPRRevision activity failures
+	for pr, future := range futureByPullNum {
+		var resp activities.SetPRRevisionResponse
+		err := future.Get(ctx, &resp)
+		if err != nil {
+			logger.Error(ctx, "error making api call to set pr revision", key.ErrKey, err, "pull_num", pr.Number)
+		}
+	}
+
+	return nil
+}
+
+func shouldRebasePullRequest(root terraform.Root, modifiedFiles []string) (bool, error) {
+	var whenModifiedRelToRepoRoot []string
+	for _, wm := range root.WhenModified {
+		wm = strings.TrimSpace(wm)
+		// An exclusion uses a '!' at the beginning. If it's there, we need
+		// to remove it, then add in the project path, then add it back.
+		exclusion := false
+		if wm != "" && wm[0] == '!' {
+			wm = wm[1:]
+			exclusion = true
+		}
+
+		// Prepend project dir to when modified patterns because the patterns
+		// are relative to the project dirs but our list of modified files is
+		// relative to the repo root.
+		wmRelPath := filepath.Join(root.Path, wm)
+		if exclusion {
+			wmRelPath = "!" + wmRelPath
+		}
+		whenModifiedRelToRepoRoot = append(whenModifiedRelToRepoRoot, wmRelPath)
+	}
+
+	// look at the filpaths for the root
+	pm, err := fileutils.NewPatternMatcher(whenModifiedRelToRepoRoot)
+	if err != nil {
+		return false, errors.Wrap(err, "building file pattern matcher using when modified config")
+	}
+
+	for _, file := range modifiedFiles {
+		match, err := pm.Matches(file)
+		if err != nil || !match {
```

Why are we just swallowing the error here?